src/components/SubscriptionsModal/SubscriptionsModal.tsx

```diff
@@ -44,14 +53,19 @@ export const SubscriptionsModal: React.FC<SubscriptionsModalProps> = ({ isModalO
   };
 
   const [selectedClientApp, setSelectedClientApp] = React.useState(
-    apps.length === 1
-      // If there's one single user app, we default the 'Client app' selection to it
-      ? apps[0]
-      // If there's more than one single user app, we leave the 'Client app' selection up to the user
-      : initialClientApp,
+    appFromRedirect
+    // If we've been redirected from the 'Applications modal', we default the 'Client app' selection to it
+      ? appFromRedirect
+      : apps.length === 1
+        // If there's one single user app, we default the 'Client app' selection to it
+        ? apps[0]
+        // If there's more than one single user app, we leave the 'Client app' selection up to the user
+        : initialClientApp,
   );
   const [isClientAppSelected, setIsClientAppSelected] = React.useState(
-    apps.length === 1,
+    appFromRedirect
+      ? true
+      : apps.length === 1,
```

Comment:
Why not use an `or` for this?

```suggestion
    !!appFromRedirect || apps.length === 1,
```